### PR TITLE
Fix broken links related to resources named "index"

### PR DESCRIPTION
**Pull request by sean1588**
fixes: https://github.com/pulumi/registry/issues/951

This PR fixes an issue where a resource named "index" has trouble resolving the url likely because "index" is some sort of reserved keyword. This change prepends a double underscore to the resource name in the URL that gets generated and also updates the links that reference this on both the module page and the package tree in the left nav. Visually everything looks the same, it is just the directory name and underlying links that get updated. It will still display as "Index" when rendered on the page.

As an example this route `/registry/packages/gcp/api-docs/firestore/index/` ends up resulting in a 404. With this change we add a double underscore which results in the following route `registry/packages/gcp/api-docs/firestore/__index/` and is now able to render the page.

Peek 2023-01-20 12-00

### CODE REVIEW

**Review by RobbieMcKinstry — APPROVED** [review]
image

This isn't a pretty edge case. I can totally see how this breaks on the frontend. No other workarounds immediately come to mind. e.g. capitalizing `Index` doesn't work particularly well. Thanks for making this change Sean!

File: pkg/codegen/docs/gen.go [hunk 1]
```diff
@@ -1771,7 +1771,16 @@ func (mod *modContext) gen(fs codegen.Fs) error {
...
 		}
 
-		addFile(strings.ToLower(title), buffer.String())
+		resourceFileName := strings.ToLower(title)
+		// Handle file generation for resources named `index`. We prepend a double underscore
+		// here, since this ends up resulting in route of .../<module>/index which has trouble
+		// resolving and returns a 404 in the browser, likely due to `index` being some sort
+		// of reserved keyword.
+		if resourceFileName == "index" {
+			resourceFileName = "__index"
```
Comment (susanev) [inline pkg/codegen/docs/gen.go:1780]: one question? could we possibly use hyphens instead of underscores? underscores are no longer supposed to be used in URLs. no worries if its not possible though.

Comment (sean1588) [reply pkg/codegen/docs/gen.go:1780]: oh thanks. never knew that. ok i'll try that out and see if it works.

Comment (sean1588) [reply pkg/codegen/docs/gen.go]: ok using the hyphens does work :tada:

**Comment by sean1588** [comment]
thanks @RobbieMcKinstry - yeah very much an edge case haha

> It looks like you'll have to regenerate the codegen tests with PULUMI_ACCEPT

yeah I ran that before I pushed and it didnt generate any new files. however for some strange reason when I just pushed the change to switch to hyphens all the tests are now passing. lol. maybe it was some sort of transient failure.

**Comment by sean1588** [comment]
bors try
